### ISSUE TITLE
Watertight mesh

### ISSUE THREAD

**Issue by feifanphys** [mikedh/trimesh#1930]
Hello,

I exported the default cube from Blender to an obj file triangulated. However, when loaded with trimesh, the cube is not watertight. The obj has 6 vertexes and trimesh shows 24. The following is the obj file I'm using.

# Blender 3.5.1
# www.blender.org
mtllib cube_tri.mtl
o Cube
v 1.000000 1.000000 -1.000000
v 1.000000 -1.000000 -1.000000
v 1.000000 1.000000 1.000000
v 1.000000 -1.000000 1.000000
v -1.000000 1.000000 -1.000000
v -1.000000 -1.000000 -1.000000
v -1.000000 1.000000 1.000000
v -1.000000 -1.000000 1.000000
vn -0.0000 1.0000 -0.0000
vn -0.0000 -0.0000 1.0000
vn -1.0000 -0.0000 -0.0000
vn -0.0000 -1.0000 -0.0000
vn 1.0000 -0.0000 -0.0000
vn -0.0000 -0.0000 -1.0000
vt 0.625000 0.500000
vt 0.375000 0.500000
vt 0.625000 0.750000
vt 0.375000 0.750000
vt 0.875000 0.500000
vt 0.625000 0.250000
vt 0.125000 0.500000
vt 0.375000 0.250000
vt 0.875000 0.750000
vt 0.625000 1.000000
vt 0.625000 0.000000
vt 0.375000 0.000000
vt 0.375000 1.000000
vt 0.125000 0.750000
s 0
usemtl Material
f 5/5/1 3/3/1 1/1/1
f 3/3/2 8/13/2 4/4/2
f 7/11/3 6/8/3 8/12/3
f 2/2/4 8/14/4 6/7/4
f 1/1/5 4/4/5 2/2/5
f 5/6/6 2/2/6 6/8/6
f 5/5/1 7/9/1 3/3/1
f 3/3/2 7/10/2 8/13/2
f 7/11/3 5/6/3 6/8/3
f 2/2/4 4/4/4 8/14/4
f 1/1/5 3/3/5 4/4/5
f 5/6/6 1/1/6 2/2/6

**Comment by Kiord** [comment]
There are 8 vertices, 6 normals, 14 texture coordinates.
In Trimesh, the attributes are stored per vertex, so when a vertex has multiple attributes, it is duplicated.

In the face construction, each vertex is associated to 3 different vertex normals. To store the correct vertex normals, you need 8x3=24 vertices.

**Comment by mikedh** [comment]
Yeah, and to combine the vertices with normals and texture you'd need to run `mesh.process(merge_tex=True, merge_norm=True)` for it to show up as watertight.
